app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java

```diff
@@ -962,6 +962,12 @@ public void openExternalUrl(Intent intent) {
   }
 
   private void externalLinkPopup(Intent intent) {
+    int warningResId;
+    if (sharedPreferenceUtil.nightMode()) {
+      warningResId = R.drawable.ic_warning_white;
```

Comment:
Adding if-else everywhere in the code will unnecessarily complicate the code. You should move this to XML using this [answer](https://stackoverflow.com/a/13735919/7792129).

Reply:
interesting, @prajurock what you think about it?